db/write_batch.cc

```diff
@@ -1002,6 +1000,9 @@ class MemTableInserter : public WriteBatch::Handler {
         SnapshotImpl read_from_snapshot;
         read_from_snapshot.number_ = sequence_;
         ReadOptions ropts;
+        // it's going to be overwritten for sure, so no point caching data block
+        // containing the old version
+        ropts.fill_cache = false;
```

Comment:
wait, I think we cache the whole data block, not just the over-written key, so in general it still make sense to cache it? (for the use case we are looking at it make sense not to cache though)

Reply:
Right we cache the whole data block. I guess it could help if the neighbors are accessed soon afterwards. But this is inside a `Put()` operation, and I don't think caching SST file blocks should be an expected outcome of the `Put()`.

Reply:
There could be use cases that caching make sense. E.g. a write only workload with inplace_update callback enable, the cache will help to reduce IO for the get query. Maybe we can make it an option in the future.